index.bs

```diff
@@ -8062,53 +8062,72 @@ IDL [=promise type=] values are represented by ECMAScript {{ECMAScript/PromiseCa
-        1.  If |T| is {{void}}, then:
-            1.  Return the result of performing any steps that were required to be run if the promise was fulfilled.
-        1.  Otherwise, |T| is a type other than {{void}}:
-            1.  Let |V| be the first argument to |onFulfilled|.
-            1.  Let |value| be the result of [=converted to an IDL value|converting=]
-                |V| to an IDL value of type |T|.
-            1.  If there are no steps that are required to be run if the promise was fulfilled, then
-                return <emu-val>undefined</emu-val>.
-            1.  Otherwise, return the result of performing any steps that were required to be run if the promise was fulfilled,
-                with |value| as the promise’s value.
+    1.  Let |onFulfilledSteps| be the following steps given argument |V|:
+        1.  Let |value| be the result of [=converted to an IDL value|converting=] |V| to an IDL
+            value of type |T|.
+        1.  If there are no steps that are required to be run if the promise was fulfilled, then
+            return <emu-val>undefined</emu-val>.
+        1.  Let |result| be the result of performing any steps that were required to be run if
+            the promise was fulfilled, given |value| if |T| is not {{void}}.
+        1.  Return |result|, [=converted to an ECMAScript value=].
     1.  Let |onFulfilled| be [=!=] [$CreateBuiltinFunction$](|onFulfilledSteps|, « »):
-    1.  Let |onRejectedSteps| be the following steps:
-        1.  Let |R| be the first argument to |onRejected|.
+    1.  Let |onRejectedSteps| be the following steps given argument |R|:
         1.  Let |reason| be the result of [=converted to an IDL value|converting=]
             |R| to an IDL value of type {{any}}.
         1.  If there are no steps that are required to be run if the promise was rejected, then
             return <emu-val>undefined</emu-val>.
-        1.  Otherwise, return the result of performing any steps that were required to be run if the promise was rejected,
-            with |reason| as the rejection reason.
+        1.  Let |result| be the result of performing any steps that were required to be run if the
+            promise was rejected, given |reason|.
+        1.  Return |result|, [=converted to an ECMAScript value=].
     1.  Let |onRejected| be [=!=] [$CreateBuiltinFunction$](|onRejectedSteps|, « »):
     1.  Let |constructor| be |promise|.\[[Promise]].\[[Realm]].\[[Intrinsics]].[[{{%Promise%}}]].
     1.  Let |newCapability| be ? [$NewPromiseCapability$](|constructor|).
     1.  Return [=!=] [$PerformPromiseThen$](|promise|.\[[Promise]], |onFulfilled|, |onRejected|, |newCapability|).
 
     Note: If the provided steps return a value of type |T|, this algorithm returns a
-    <a interface lt="Promise">Promise&lt;|T|&gt;</a>.
+    <code>{{Promise}}<var ignore>T</var>&gt;</code>.
+</div>
+
+<div algorithm>
+    To perform some steps <dfn export for=Promise>upon fulfillment</dfn> of a
```

Comment:
Realistically I think most of these don't need for=Promise; there's no competition to disambiguate with. At least, not the ones that explicitly have "promise" in their name, for sure...